wdl2cwl/main.py

```diff
@@ -36,784 +21,766 @@
+                            f"$({placeholder_expr}.map("
+                            + 'function(el) {return el.path}).join("'
+                            + seperator
+                            + '"))'
+                        )
+                else:
+                    raise Exception(
+                        f"{wdl_placeholder} with expr of type {expr.type} is not yet handled"
+                    )
             else:
-                ram_min = get_ram_min(ast.task_runtime["memory"])
+                raise Exception(
+                    f"Placeholders with options {options} are not yet handled."
+                )
         else:
-            ram_min = get_ram_min_js(ast.task_runtime["memory"], "")
-
-        requirements.append(
-            cwl.ResourceRequirement(
-                ramMin=ram_min,
+            # for the one case where the $(input.some_input_name) is used within the placeholder_expr
+            # we return the placholder_expr without enclosing in another $()
+            cwl_command_str = (
+                f"$({placeholder_expr})"
+                if placeholder_expr[-1] != ")"
+                else placeholder_expr
             )
+        # sometimes placeholders are used inside WDL.Expr.String.
+        # with the parent and grand_parent we can confirm that we are in
+        # the command string (WDL.Expr.String) and task (WDL.Tree.Task) respectively
+        parent = wdl_placeholder.parent  # type: ignore
+        grand_parent = parent.parent
+        return (
+            cwl_command_str
+            if isinstance(parent, WDL.Expr.String)
+            and isinstance(grand_parent, WDL.Tree.Task)
+            else cwl_command_str[2:-1]
         )
 
-    outdir_min = get_outdir_min(ast.task_runtime.get("disks", "1"))
-    requirements.append(cwl.ResourceRequirement(outdirMin=outdir_min))
-
-    if "time_minutes" in ast.task_runtime:
-
-        time_minutes: Union[str, int] = ""
-        if '"' not in ast.task_runtime["time_minutes"]:
-            if ast.task_runtime["time_minutes"] in input_names:
-                time_minutes = f"$({inputs(ast.task_runtime['time_minutes'])} * 60)"
-            elif ast.task_runtime["time_minutes"].isnumeric():
-                time_minutes = int(ast.task_runtime["time_minutes"]) * 60
-        else:
-            print(
-                f"---- WARNING: SKIPPING REQUIREMENT time_minutes in  {ast.task_runtime} ----",
-                file=sys.stderr,
-            )
-
-        requirements.append(
-            cwl.ToolTimeLimit(
-                timelimit=time_minutes,
-            )
-        )
+    def get_wdl_literal(
+        self, wdl_expr: Union[WDL.Expr.Int, WDL.Expr.Float, WDL.Expr.Boolean]
+    ) -> Any:
+        """Extract Literal value from WDL expr."""
+        if wdl_expr is None or not hasattr(wdl_expr, "value"):
+            raise Exception(f"{type(wdl_expr)} has not attribute 'value'")
+        value = wdl_expr.value
+        return value
```

Comment:
All of the possible inputs have a `value` attribute; why do we check for it? Can't this function be removed and instead accessing the  `.value` attribute directly?